internal/controller/resource_manager.go

```diff
@@ -31,38 +30,39 @@ type ResourceManager struct {
 
 // TODO: Make a struct with the ClientSet as a field, and convertMetaObjectToMetadata, getPodIPAddresses, getProviderIdNodeSpec should be methods of that struct.
 
-// DynamicListAndWatchResources lists and watches the specified resource
-// dynamically, managing context cancellation and synchronization with wait
-// groups. As long as we are able to watch and stream, we will never return from
-// this function
-func (r *ResourceManager) DynamicListAndWatchResources(ctx context.Context, cancel context.CancelFunc, resource string, apiGroup string, allResourcesSnapshotted *sync.WaitGroup, snapshotCompleted *sync.WaitGroup) {
-	defer cancel()
-
-	resourceListVersion, err := r.DynamicListResources(ctx, r.logger, resource, apiGroup)
+// ListK8sResources lists the current state of a specified Kubernetes resource
+// using the dynamic client, and returns its latest resourceVersion.
+func (r *ResourceManager) ListK8sResources(ctx context.Context, resource string, apiGroup string) (string, error) {
+	resourceVersion, err := r.DynamicListResources(ctx, r.logger, resource, apiGroup)
 	if err != nil {
-		allResourcesSnapshotted.Done()
-		r.logger.Error("Unable to list resources", zap.Error(err))
-		return
+		r.logger.Error("Failed to list resource", zap.String("resource", resource), zap.Error(err))
```

Comment:
All this method does is wrap `DynamicListResources` to log the error if it's not `nil`.
That error is logged by the caller with the exact same log message here: https://github.com/illumio/cloud-operator/pull/180/files#diff-65d4066e2022c168a97a2083474d348fa6d5d1fffb6e8d7378282935ad187940R204

So this `ListK8sResources` function is unnecessary. Just have the caller call `DynamicListResources` directly.